azurerm/resource_arm_application_insights_webtests.go

```diff
@@ -0,0 +1,314 @@
+	testConf := d.Get("test_configuration").(string)
+
+	tags := d.Get("tags").(map[string]interface{})
+	tagKey := fmt.Sprintf("hidden-link:/subscriptions/%s/resourceGroups/%s/providers/microsoft.insights/components/%s", client.SubscriptionID, resGroup, appInsightsName)
+	tags[tagKey] = "Resource"
+
+	testConfiguration := insights.WebTestPropertiesConfiguration{
+		WebTest: &testConf,
+	}
+
+	webTestProperties := insights.WebTestProperties{
+		SyntheticMonitorID: &name,
+		WebTestName:        &name,
+		Description:        &description,
+		Enabled:            &isEnabled,
+		Frequency:          &frequency,
+		Timeout:            &timeout,
+		WebTestKind:        insights.WebTestKind(kind),
+		RetryEnabled:       &retryEnabled,
+		Locations:          &geoLocations,
+		Configuration:      &testConfiguration,
+	}
+
+	webTest := insights.WebTest{
+		Name:              &name,
+		Location:          &location,
+		Kind:              insights.WebTestKind(kind),
+		WebTestProperties: &webTestProperties,
+		Tags:              expandTags(tags),
+	}
+
+	resp, err := client.CreateOrUpdate(ctx, resGroup, name, webTest)
+	if err != nil {
+		return fmt.Errorf("Error creating Application Insights WebTest %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmApplicationInsightsWebTestsRead(d, meta)
+}
+
+func resourceArmApplicationInsightsWebTestsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appInsightsWebTestsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Reading AzureRM Application Insights WebTests '%s'", id)
+
+	resGroup := id.ResourceGroup
+	name := id.Path["webtests"]
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Application Insights WebTest %q was not found in Resource Group %q - removing from state!", name, resGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving Application Insights WebTests %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	appInsightsId := ""
+	tags := resp.Tags
+	for i := range tags {
+		if strings.HasPrefix(i, "hidden-link") {
+			appInsightsId = strings.Split(i, ":")[1]
+		}
+	}
+	d.Set("application_insights_id", appInsightsId)
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("kind", resp.Kind)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if props := resp.WebTestProperties; props != nil {
+		d.Set("synthetic_monitor_id", props.SyntheticMonitorID)
+		d.Set("description", props.Description)
+		d.Set("enabled", props.Enabled)
+		d.Set("frequency", props.Frequency)
+		d.Set("timeout", props.Timeout)
+		d.Set("retry_enabled", props.RetryEnabled)
+		d.Set("configuration", props.Configuration)
```

Comment:
This is not in the schema?

Reply:
that one covered by the lines below - removed